lib/_http_common.js

```diff
@@ -126,12 +126,6 @@ function parserOnMessageComplete() {
       parser._url = '';
     }
 
-    if (!stream.upgrade)
-      // For upgraded connections, also emit this after parser.execute
-      stream.push(null);
-  }
-
-  if (stream && !parser.incoming._pendings.length) {
```

Comment:
This is the only place I could find where `._pendings` was actually used, so since nothing was ever added to the array, the second part of the conditional was always evaluating to true. Thus, the `stream.push(null)` can be merged up above.